contracts/upgradeable_contracts/stake_token_mediators/ForeignStakeTokenMediator.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity 0.4.24;
+
+import "./BasicStakeTokenMediator.sol";
+import "../../interfaces/IBurnableMintableERC677Token.sol";
+
+contract ForeignStakeTokenMediator is BasicStakeTokenMediator {
+    /**
+     * @dev Executes action on incoming tokens from the other network
+     * @param _recipient address of tokens receiver
+     * @param _value amount of bridged tokens
+     */
+    function executeActionOnBridgedTokens(address _recipient, uint256 _value) internal {
+        uint256 value = _value.div(10**decimalShift());
```

Comment:
Do we need to include such check in `ambErcToErc` mediators also?
Is it sufficient to check this condition only once, on one side of a bridge?

Reply:
The question is about the unit tests. Do we have such test? Or the answer is 'no' and moreover there is no such limitation that is why you are requesting for requirements?

Reply:
I don't see the place where we are validating value to be non-zero

Reply:
So, does it mean that there is no such check neither for the `native-to-erc20` mode nor for `relayTokens` functionality for `erc20-to-native`/`erc20-to-erc20` modes and we are able to request transfer of `0` tokens?

Reply:
If I understand it correctly, the current situation is the following:
* In `relayTokens` for `erc20-to-native`/`erc20-to-erc20`, we make such check.
* In regular ERC20 transfer for `native-to-erc20`, `erc20-to-native` and `erc20-to-erc20`, we cannot prevent token contract to execute a call with zero amount.
* In `ambErcToErc` we use ERC677 token, so it is possible to revert in `onTokenTransfer`. We don't check if amount is positive there, BUT we check that amount is within the limits. The `minPerTx` limit cannot be set to zero, that's why, I guess, we don't have an explicit statement `require(value > 0)`.

I made a manual test on deployed test STAKE bridge, it does not allow to transfer 0 tokens to the mediator. Do we need an additional explicit test for zero value, if we already have tests for values which are out of limits?

Reply:
> In regular ERC20 transfer for native-to-erc20, erc20-to-native and erc20-to-erc20, we cannot prevent token contract to execute a call with zero amount.

Hmm... Since for the case `native-to-erc20` we are using the ERC677 token contract, the `transfer` method will eventually executes `onTokenTransfer` that will check whether the value is within the limits.

But `erc20-to-native` and `erc20-to-erc20` do not have such check, you are right. For them, the following check https://github.com/poanetwork/tokenbridge-contracts/blob/5f57e8ff8ee887c9fad0518c7aa58aaf91cde88e/contracts/upgradeable_contracts/BasicHomeBridge.sol#L46 will make sure that `0` will not be propagated further (e.g. to the block reward contract).

> Do we need an additional explicit test for zero value

Yes, it makes sense.